HADOOP-19674. [JDK 17] Add jaxb-runtime

K0K0V0K opened the pull request:
https://issues.apache.org/jira/browse/HADOOP-19674

### Description of PR

- Since JDK11 javax.xml.bind modules is removed
    source: https://docs.oracle.com/en/java/javase/24/migrate/removed-tools-and-components.html#GUID-11F78105-D735-430D-92DD-6C37958FCBC3
- Jetty needs a jaxb-impl to be able to start
- So we provide the jaxb-runtime artifact everywhere where Jetty Server is a dependency
- If some modules excludes jetty-server then most probably jaxb-runtime is not needed for them, so we exclude it.
- ehcache was using different version of jaxb-runtime, so it become excluded

### How was this patch tested?

Unit tests
- I created a JDK8 build and run the `org.apache.hadoop.yarn.webapp.TestWebApp` test
- I created a JDK17 build and run the` org.apache.hadoop.yarn.webapp.TestWebApp` test

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

stoty commented on hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/pom.xml:
```diff
@@ -122,6 +122,12 @@
+    <dependency>
+      <groupId>org.glassfish.jaxb</groupId>
+      <artifactId>jaxb-runtime</artifactId>
+      <version>${jaxb.version}</version>
```
The version should be set in once in the hadoop-project pom dependencyManagement section

stoty commented on hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/pom.xml:
```diff
@@ -144,6 +144,11 @@
       <artifactId>jersey-media-jaxb</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
How did you decide where to add jaxb-imp and with what scope ?

K0K0V0K replied:
I checked the usage of the `JettisonJaxbContext` in the projects where only used in the tests i added with test scope, otherwise i set no scope.

stoty commented on hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/pom.xml:
```diff
@@ -208,6 +208,12 @@
+      <groupId>org.glassfish.jaxb</groupId>
+      <artifactId>jaxb-runtime</artifactId>
+      <version>${jaxb.version}</version>
+      <scope>test</scope>
```
This is needed by Jetty. (at least)
This should be compile scope, as javax.xml.bind is directly used in the main code.

K0K0V0K replied:
I was not sure this needed in every case for the Jetty.
But in that case if i see well we will need this every where where we are using the HttpServer2.
So if i see well this will be needed in HDFS NN also.

In that case shall we add this to the `hadoop-common` project only and let others use it transitively?

stoty replied:
> I was not sure this needed in every case for the Jetty.

The stack strace seems to be coming from Jetty initialization.

> But in that case if i see well we will need this every where where we are using the HttpServer2. So if i see well this will be needed in HDFS NN also.

Yes, that's likely.

> 
> In that case shall we add this to the `hadoop-common` project only and let others use it transitively?

I think it's better to add it separately where needed.
Most modules don't use Jetty.

stoty replied:
One other issue is the binary assembly

We also need to make sure that jaxb-impl is included there.

K0K0V0K replied:
I tried to upload a corrected version.
With my understanding every Hadoop based Jetty server is instance of the `HttpServer2`, what is in the `hadoop-common`, so i added the dependency there.

Some other modules what use` hadoop-common`, but not the Jetty server are already excludes the `jetty-server`.
I extended these cases with `jaxb-runtime` exculsion. This means: not 100% these modules dont need `jaxb-runtime`, but pretty sure not because the Jetty.

In the httpfs modul i dont know why but seems like the `jetty-server` transitive dependency is excluded from `hadoop-common`, but there is an explicit dependency for it also in the pom.

I run a clean install with -Pdist and i can see the jaxb-runtime jar in the `hadoop-dist/target/hadoop-3.5.0-SNAPSHOT/share/hadoop/common/lib/jaxb-runtime-2.3.9.jar` path

susheelgupta7 commented:
Build issue observed on JDK 8
The Maven enforcer plugin failed due to a dependency convergence conflict for jaxb-runtime:

```java
[ERROR] Rule 0: org.apache.maven.enforcer.rules.dependency.DependencyConvergence failed with message:
[ERROR] Failed while enforcing releasability.
[ERROR] 
[ERROR] Dependency convergence error for org.glassfish.jaxb:jaxb-runtime:jar:2.3.1 paths to dependency are:
[ERROR] +-org.apache.hadoop:hadoop-mapreduce-client-app:jar:3.5.0-SNAPSHOT
[ERROR]   +-org.apache.hadoop:hadoop-yarn-server-web-proxy:jar:3.5.0-SNAPSHOT:compile
[ERROR]     +-org.apache.hadoop:hadoop-yarn-server-common:jar:3.5.0-SNAPSHOT:compile
[ERROR]       +-org.ehcache:ehcache:jar:3.8.2:compile
[ERROR]         +-org.glassfish.jaxb:jaxb-runtime:jar:2.3.1:compile
[ERROR] and
[ERROR] +-org.apache.hadoop:hadoop-mapreduce-client-app:jar:3.5.0-SNAPSHOT
[ERROR]   +-org.glassfish.jaxb:jaxb-runtime:jar:2.3.9:compile
```

K0K0V0K commented:
> Build issue observed on JDK 8 The Maven enforcer plugin failed due to a dependency convergence conflict for jaxb-runtime:
> 
> ```java
> [ERROR] Rule 0: org.apache.maven.enforcer.rules.dependency.DependencyConvergence failed with message:
> [ERROR] Failed while enforcing releasability.
> [ERROR] 
> [ERROR] Dependency convergence error for org.glassfish.jaxb:jaxb-runtime:jar:2.3.1 paths to dependency are:
> [ERROR] +-org.apache.hadoop:hadoop-mapreduce-client-app:jar:3.5.0-SNAPSHOT
> [ERROR]   +-org.apache.hadoop:hadoop-yarn-server-web-proxy:jar:3.5.0-SNAPSHOT:compile
> [ERROR]     +-org.apache.hadoop:hadoop-yarn-server-common:jar:3.5.0-SNAPSHOT:compile
> [ERROR]       +-org.ehcache:ehcache:jar:3.8.2:compile
> [ERROR]         +-org.glassfish.jaxb:jaxb-runtime:jar:2.3.1:compile
> [ERROR] and
> [ERROR] +-org.apache.hadoop:hadoop-mapreduce-client-app:jar:3.5.0-SNAPSHOT
> [ERROR]   +-org.glassfish.jaxb:jaxb-runtime:jar:2.3.9:compile
> ```

Thanks @susheelgupta7 for the finding!

K0K0V0K committed:
HADOOP-19674. [JDK 17] Add jaxb-runtime

- Since JDK11 javax.xml.bind modules is removed
    source: https://docs.oracle.com/en/java/javase/24/migrate/removed-tools-and-components.html#GUID-11F78105-D735-430D-92DD-6C37958FCBC3
- Jetty needs a jaxb-impl to be able to start
- So we provide the jaxb-runtime artifact everywhere where Jetty Server is a dependency
- If some modules excludes jetty-server then most probably jaxb-runtime is not needed for them, so we exclude it.
- ehcache was using different version of jaxb-runtime, so it become excluded

stoty commented on hadoop-client-modules/hadoop-client/pom.xml:
```diff
@@ -53,6 +53,10 @@
           <groupId>org.eclipse.jetty</groupId>
           <artifactId>jetty-server</artifactId>
         </exclusion>
+        <exclusion>
```
These are for java 8, right ?

stoty commented on hadoop-project/pom.xml:
```diff
@@ -2152,6 +2155,11 @@
         <artifactId>jersey-media-json-jettison</artifactId>
         <version>${jersey2.version}</version>
       </dependency>
+      <dependency>
```
Since we're globally dependency managing jaxb-runtime, I think that we don't need all the excludes in this patch.
They should all be dependency managed to 2.3.9 by maven, and the dependency convergence check should pass.

stoty replied:
I think that the reported error was for the earlier patch without the global dependency management.

stoty reviewed (commented):
Please check if the new exclusions are really needed.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   9m 20s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  38m 12s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 54s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 43s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |   8m  2s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   7m 13s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   6m 15s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  | 150m  1s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 40s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 24s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 21s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 30s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  16m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   7m 41s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   7m  7s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   6m 11s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  65m 10s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 31s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |  24m  7s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 52s |  |  hadoop-kms in the patch passed.  |
| +1 :green_heart: |  unit  |   6m  0s |  |  hadoop-hdfs-httpfs in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 16s |  |  hadoop-yarn-server-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 35s |  |  hadoop-yarn-server-web-proxy in the patch passed.  |
| -1 :x: |  unit  | 145m 56s | [/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/2/artifact/out/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client.txt) |  hadoop-mapreduce-client in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 38s |  |  hadoop-yarn-services-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 53s |  |  hadoop-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 39s |  |  The patch does not generate ASF License warnings.  |
|  |   | 451m 50s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.mapreduce.v2.TestUberAM |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint |
| uname | Linux 4eb1be1e663a 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 6086bd5ba19b4e4de56b0eba69be70962117a85a |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/2/testReport/ |
| Max. process+thread count | 1273 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

K0K0V0K committed:
HADOOP-19674. [JDK 17] Add jaxb-runtime

- Remove not necessary excludes

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   9m  4s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  42m 58s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 38s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m 35s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  24m 49s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  11m  2s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   9m 33s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  62m 19s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 37s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  42m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  19m 10s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  19m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 22s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  16m 22s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  21m 21s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |  11m  2s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   8m 13s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  62m  4s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 488m 32s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/3/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 15s |  |  The patch does not generate ASF License warnings.  |
|  |   | 831m 35s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.TestRollingUpgrade |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux a9562a3960c9 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 450d122aa0923580d4ddfe2b8ad9a734265fe7d9 |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/3/testReport/ |
| Max. process+thread count | 2197 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/3/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

K0K0V0K committed:
HADOOP-19674. [JDK 17] Add jaxb-runtime

- Put back one exclude what seems to be necessary
- Otherwise at and of install we can see the following error:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (enforce-banned-dependencies) on project hadoop-client-check-test-invariants:
[ERROR] Rule 1: org.apache.maven.plugins.enforcer.BanDuplicateClasses failed with message:
[ERROR] Duplicate classes found:
[ERROR]
[ERROR]   Found in:
[ERROR]     org.apache.hadoop:hadoop-client-minicluster:jar:3.5.0-SNAPSHOT:compile
[ERROR]     org.apache.hadoop:hadoop-client-runtime:jar:3.5.0-SNAPSHOT:compile
[ERROR]   Duplicate classes:
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MimeTypeEntry.class
...

Cannon to right of them,

K0K0V0K commented on hadoop-client-modules/hadoop-client/pom.xml:
```diff
@@ -53,6 +53,10 @@
+        <exclusion>
+          <groupId>org.glassfish.jaxb</groupId>
+          <artifactId>jaxb-runtime</artifactId>
+        </exclusion>
```
I think this one is necessary otherwise at the end on install i am facing this error:

```
[INFO] Total time:  05:20 min (Wall Clock)
[INFO] Finished at: 2025-09-05T09:02:10+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (enforce-banned-dependencies) on project hadoop-client-check-test-invariants: 
[ERROR] Rule 1: org.apache.maven.plugins.enforcer.BanDuplicateClasses failed with message:
[ERROR] Duplicate classes found:
[ERROR] 
[ERROR]   Found in:
[ERROR]     org.apache.hadoop:hadoop-client-minicluster:jar:3.5.0-SNAPSHOT:compile
[ERROR]     org.apache.hadoop:hadoop-client-runtime:jar:3.5.0-SNAPSHOT:compile
[ERROR]   Duplicate classes:
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MimeTypeEntry.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/LocalizableMessageFactory$ResourceBundleSupplier.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MailcapTokenizer.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/FinalArrayList.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/Pool$Impl.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MailcapParseException.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/LogSupport.class
[ERROR]     META-INF/versions/9/com/sun/istack/logging/StackHelper.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/NotNull.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/XMLStreamException2.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/viewers/TextEditor.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/XMLStreamReaderToContentHandler$1.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/Localizable.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/viewers/ImageViewerCanvas.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/viewers/TextViewer.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/Interned.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/Localizer.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/LocalizableMessageFactory.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MailcapFile.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/logging/Logger.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/ByteArrayDataSource.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/Nullable.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/NullLocalizable.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/SAXException2.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/MimeTypeFile.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/logging/StackHelper.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/registries/LineTokenizer.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/Builder.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/FragmentContentHandler.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/Pool.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/localization/LocalizableMessage.class
[ERROR]     org/apache/hadoop/shaded/com/sun/activation/viewers/ImageViewer.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/SAXParseException2.class
[ERROR]     org/apache/hadoop/shaded/com/sun/istack/XMLStreamReaderToContentHandler.class
```

stoty replied:
I have checked 3.4.2, and these classes are not present there.
Based on that, it seems that hadoop-client-runtime indeed does not include jaxb-impl traditionally.

stoty reviewed (approved):
+1 LGTM

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  15m 34s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   8m 17s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   5m 15s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/7/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  compile  |  17m 32s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  14m 30s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  23m 18s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  10m  7s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   8m 13s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  56m  8s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 37s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  36m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  1s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m  1s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 40s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  14m 40s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  22m 13s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |  10m 38s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   8m 20s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  61m 21s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 446m 39s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/7/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   | 744m 47s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux a22a22156e0a 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / c5a26a7263c9a5c356216b6426d5fe903213cd15 |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/7/testReport/ |
| Max. process+thread count | 3607 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/7/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 56s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 56s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  40m 57s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 53s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m 46s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  23m 46s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  10m 34s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 51s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  58m 27s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 37s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  37m 21s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  compile  |  17m 10s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 11s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  15m 11s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  21m  2s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   9m 54s | [/patch-javadoc-root-jdkUbuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/artifact/out/patch-javadoc-root-jdkUbuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javadoc  |   7m 32s | [/patch-javadoc-root-jdkPrivateBuild-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/artifact/out/patch-javadoc-root-jdkPrivateBuild-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.txt) |  root in the patch failed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.  |
| -1 :x: |  shadedclient  |  61m 30s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 480m  5s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 14s |  |  The patch does not generate ASF License warnings.  |
|  |   | 810m  1s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.ha.TestStandbyCheckpoints |
|   | hadoop.hdfs.server.namenode.TestNameNodeMXBean |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux bd7276601e6c 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 835934188aaa69fac727bbe178075cadfc87e0a9 |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/testReport/ |
| Max. process+thread count | 2322 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/4/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 51s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 51s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  37m 23s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 47s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  17m 16s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  24m  2s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  11m 10s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   8m 52s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  59m 48s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  35m 24s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  compile  |  18m  7s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m  7s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 15s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  17m 15s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  22m 30s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  |  No new issues.  |
| -1 :x: |  javadoc  |  11m 51s | [/patch-javadoc-root-jdkUbuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/artifact/out/patch-javadoc-root-jdkUbuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javadoc  |   8m 21s | [/patch-javadoc-root-jdkPrivateBuild-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/artifact/out/patch-javadoc-root-jdkPrivateBuild-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.txt) |  root in the patch failed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09.  |
| -1 :x: |  shadedclient  |  63m  0s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 611m 20s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 18s |  |  The patch does not generate ASF License warnings.  |
|  |   | 944m 54s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux f7e642f299bc 5.15.0-151-generic #161-Ubuntu SMP Tue Jul 22 14:25:40 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 835934188aaa69fac727bbe178075cadfc87e0a9 |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/testReport/ |
| Max. process+thread count | 3138 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/5/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

K0K0V0K committed:
HADOOP-19674. [JDK 17] Add jaxb-runtime

- If we exclude the dependency from hadoop common, maybe we should do the same with the hdfs-client

Cannon to left of them,

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  22m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 46s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  37m 36s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 44s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m 22s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  23m 15s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  10m 21s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 43s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  58m 26s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 37s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  41m  8s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 44s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 13s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  15m 13s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  20m 23s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |  10m  6s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 45s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  59m 47s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 481m 54s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/8/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 15s |  |  The patch does not generate ASF License warnings.  |
|  |   | 824m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux 117987a87daa 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 7d1831a9a0ae7b1f73edfce600e8a818b5ec5de6 |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/8/testReport/ |
| Max. process+thread count | 3135 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/8/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 commented:
@K0K0V0K Thank you for your contribution! LGTM. However, we need to trigger the compilation again as the current process seems to have been interrupted.

I also noticed some warning messages. Could you please double-check?


```
[WARNING] 

Some dependencies of Maven Plugins are expected to be in provided scope.
Please make sure that dependencies listed below declared in POM
have set '<scope>provided</scope>' as well.

The following dependencies are in wrong scope:
 * org.apache.maven:maven-plugin-api:jar:3.9.5:compile
 * org.apache.maven:maven-model:jar:3.9.5:compile
 * org.apache.maven:maven-artifact:jar:3.9.5:compile
 * org.apache.maven:maven-core:jar:3.9.5:compile
 * org.apache.maven:maven-settings:jar:3.9.5:compile
 * org.apache.maven:maven-settings-builder:jar:3.9.5:compile
 * org.apache.maven:maven-builder-support:jar:3.9.5:compile
 * org.apache.maven:maven-repository-metadata:jar:3.9.5:compile
 * org.apache.maven:maven-model-builder:jar:3.9.5:compile
 * org.apache.maven:maven-resolver-provider:jar:3.9.5:compile
```

K0K0V0K commented:
Hi @slfan1989!

Thanks for the review. I checked the latest console, output and seems like the waring is not present anymore:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/8/consoleFull

K0K0V0K committed:
HADOOP-19674. [JDK 17] Add jaxb-runtime

- there were 2 empty line at end of file, so I removed one of them to retrigger the build.

Cannon in front of them.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 51s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  38m 27s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 56s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 29s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  mvnsite  |  23m  9s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  10m 16s |  |  trunk passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 46s |  |  trunk passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  58m 11s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 39s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  41m 29s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m  0s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m  0s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 40s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  javac  |  15m 40s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  19m 58s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |  10m 17s |  |  the patch passed with JDK Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 44s |  |  the patch passed with JDK Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09  |
| +1 :green_heart: |  shadedclient  |  59m 50s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 656m 49s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/9/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   | 980m 24s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.TestRollingUpgrade |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7928 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint shellcheck shelldocs |
| uname | Linux ca1384aa87d7 5.15.0-143-generic #153-Ubuntu SMP Fri Jun 13 19:10:45 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 56b01d2c3fd0ade4f966d40068b9fdc9c53dc7ff |
| Default Java | Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.27+6-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_452-8u452-ga~us1-0ubuntu1~20.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/9/testReport/ |
| Max. process+thread count | 3135 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-common-project/hadoop-kms hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-client-modules/hadoop-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7928/9/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

K0K0V0K commented:
Thanks! @stoty @slfan1989 @brumi1024 @susheelgupta7 for the review and for the help here!
